src/app/dataset-view/additional-components/flows-component/components/flows-table/flows-table.component.ts

```diff
@@ -41,22 +42,28 @@ export class FlowsTableComponent implements OnInit, OnChanges {
     @Input() public nodes: FlowSummaryDataFragment[];
     @Input() public filterByStatus: MaybeNull<FlowStatus>;
     @Input() public filterByInitiator: FilterByInitiatorEnum;
-    @Input() public searchByAccountName: string;
-    @Input() public fetchStep: MaybeUndefined<FetchStep>;
-    @Input() public transformData: TransformDescriptionTableData;
-    @Input() public datasetBasics: DatasetBasicsFragment;
+    @Input() public searchByAccount: MaybeNull<Account>;
+    @Input() public searchByDataset: MaybeNull<Dataset>;
+    @Input() tableOptions: FlowsTableOptions;
     @Output() public filterByStatusChange = new EventEmitter<MaybeNull<FlowStatus>>();
     @Output() public filterByInitiatorChange = new EventEmitter<FilterByInitiatorEnum>();
-    @Output() public searchByAccountNameChange = new EventEmitter<string>();
-    @Output() public cancelFlowChange = new EventEmitter<string>();
-    public readonly DISPLAY_COLUMNS: string[] = ["description", "information", "creator", "options"];
-    public readonly INITIATORS: string[] = Object.keys(FilterByInitiatorEnum);
+    @Output() public searchByAccountNameChange = new EventEmitter<MaybeNull<Account>>();
+    @Output() public searchByDatasetNameChange = new EventEmitter<MaybeNull<Dataset>>();
+    @Output() public cancelFlowChange = new EventEmitter<CancelFlowArgs>();
     public readonly DEFAULT_AVATAR_URL = AppValues.DEFAULT_AVATAR_URL;
     public readonly DEFAULT_FLOW_INITIATOR = AppValues.DEFAULT_FLOW_INITIATOR;
     public readonly FlowStatus: typeof FlowStatus = FlowStatus;
     public readonly FilterByInitiatorEnum: typeof FilterByInitiatorEnum = FilterByInitiatorEnum;
+    public readonly SEARCH_DATASET_TYPEAHEAD_HEIGHT = 65;
+    public readonly TYPEAHEAD_ITEM_HEIGHT = 30;
+
     public dataSource: MatTableDataSource<FlowSummaryDataFragment> = new MatTableDataSource<FlowSummaryDataFragment>();
     @ViewChildren(MatMenuTrigger) triggersMatMenu: QueryList<MatMenuTrigger>;
+    public searchingAccount: boolean = false;
+    public searchingDataset: boolean = false;
+    public searchAccountDatasetsLength: number;
+    @Input() public accountFlowInitiators: MaybeNull<Account[]> = null;
```

Comment:
Why this port about initiators is only passed in dataset parent and not in account parent?

Dataset:
`[accountFlowInitiators]="accountFlowInitiators$ | async"`

Account:
n/a

Reply:
Added